handlers/graph.go

```diff
@@ -217,69 +250,86 @@ func buildNamespaceTrafficMap(namespace string, o options.Options, client *prome
 
 		// fetch the internally originating request traffic time-series
 		intIstioVector := promQuery(query, time.Unix(o.QueryTime, 0), client.API())
-		populateTrafficMap(trafficMap, &intIstioVector, o)
+		populateTrafficMapHttp(trafficMap, &intIstioVector, o)
 	}
 
 	return trafficMap
 }
 
-func populateTrafficMap(trafficMap graph.TrafficMap, vector *model.Vector, o options.Options) {
-	for _, s := range *vector {
-		m := s.Metric
-		lSourceWlNs, sourceWlNsOk := m["source_workload_namespace"]
-		lSourceWl, sourceWlOk := m["source_workload"]
-		lSourceApp, sourceAppOk := m["source_app"]
-		lSourceVer, sourceVerOk := m["source_version"]
-		lDestSvcNs, destSvcNsOk := m["destination_service_namespace"]
-		lDestSvcName, destSvcNameOk := m["destination_service_name"]
-		lDestWl, destWlOk := m["destination_workload"]
-		lDestApp, destAppOk := m["destination_app"]
-		lDestVer, destVerOk := m["destination_version"]
-		lCode, codeOk := m["response_code"]
+func mapSample(trafficMap graph.TrafficMap, o options.Options, sample *model.Sample) (*graph.Node, *graph.Node, *graph.Edge, bool) {
+	m := sample.Metric
+	lSourceWlNs, sourceWlNsOk := m["source_workload_namespace"]
+	lSourceWl, sourceWlOk := m["source_workload"]
+	lSourceApp, sourceAppOk := m["source_app"]
+	lSourceVer, sourceVerOk := m["source_version"]
+	lDestWlNs, destSvcNsOk := m["destination_workload_namespace"]
```

Comment:
Assuming you continue to use destination_workload_namespace for TCP, and return to destination_service_namespace for HTTP, I'd suggest you eliminate this shared function and just move this back to the individual protocol methods.

Reply:
Yes, it's moved back to the methods.